src/Compilers/Core/Portable/Operations/Operation.cs

```diff
@@ -154,41 +159,64 @@ public static IOperation CreateOperationNone(SemanticModel semanticModel, Syntax
             return operations;
         }
 
-        private class NoneOperation : Operation
+        private abstract class BaseNoneOperation : Operation
         {
-            private readonly Lazy<ImmutableArray<IOperation>> _lazyChildren;
-
-            public NoneOperation(SemanticModel semanticModel, SyntaxNode node, Optional<object> constantValue, Func<ImmutableArray<IOperation>> getChildren, bool isImplicit) :
-                base(OperationKind.None, semanticModel, node, type: null, constantValue: constantValue, isImplicit: isImplicit)
-            {
-                _lazyChildren = new Lazy<ImmutableArray<IOperation>>(getChildren);
-            }
-
-            public override void Accept(OperationVisitor visitor)
+            protected BaseNoneOperation(SemanticModel semanticModel, SyntaxNode syntax, Optional<object> constantValue, bool isImplicit) :
+                base(OperationKind.None, semanticModel, syntax, type: null, constantValue, isImplicit)
             {
-                visitor.VisitNoneOperation(this);
             }
 
-            public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-            {
-                return visitor.VisitNoneOperation(this, argument);
-            }
+            protected abstract ImmutableArray<IOperation> ChildrenImpl { get; }
 
             public override IEnumerable<IOperation> Children
             {
                 get
                 {
-                    foreach (var child in _lazyChildren.Value)
+                    foreach (var child in ChildrenImpl)
                     {
                         if (child == null)
                         {
                             continue;
                         }
 
-                        yield return Operation.SetParentOperation(child, this);
+                        yield return SetParentOperation(child, this);
```

Comment:
>yield return SetParentOperation(child, this); [](start = 24, length = 45)

We should not be doing this for non-lazy flavor. Parent should be assigned to children in constructor. We got rid of the Impl pattern everywhere else